src/cmd/ksh93/tests/ifs.sh

```diff
@@ -0,0 +1,30 @@
+# These are the tests for the internal field separator (IFS).
+
+IFS=e
+set : :
+[[ "$*" == ":e:" ]] || log_error "IFS failed" ":e:" "$*"
+
+IFS='|' read -r first second third <<< 'one|two|three'
+[[ "${first}" == "one" ]] || log_error "IFS failed" "one" "${first}"
+[[ "${second}" == "two" ]] || log_error "IFS failed" "two" "${second}"
+[[ "${third}" == "three" ]] || log_error "IFS failed" "three" "${third}"
+
+# Multi-byte character checks will only work if UTF-8 inputs are enabled
+if [ "${LANG}" = "C.UTF-8" ]
+then
+    # 2 byte latin accented e character
+    IFS=é
```

Comment:
This statement raises an interesting question. All (nearly all?) of the unit tests presume to be running in an ASCII locale unless they explicitly set the relevant locale env vars to a non-ASCII locale. What does it mean to run a ksh script with a UTF-8 encoded character in a non-UTF-8 locale? Yes, your test a few lines above should keep this statement from being executed in that case. But the script will have already been parsed.j

Reply:
I actually ran this test locally and in such a case the script parses fine - although the 2-byte character naturally becomes two separate characters. It strikes me that the internal lexer and parser themselves are highly reliable on the ASCII/UTF-8 encoding to function - so I suspect deeply deviant encodings would have other problems here.